0000-catalog.md

```diff
@@ -0,0 +1,779 @@
+product schema definition.
+
+**Grid Product from the [Product
+RFC](https://github.com/hyperledger/grid-rfcs/blob/master/text/0005-product.md):**
+
+```
+message Product {
+    enum ProductNamespace {
+        UNSET_NAMESPACE = 0;
+        GS1 = 1;
+    }
+
+    ProductNamespace product_namespace = 1;
+    string product_id = 2;
+    string owner = 3;
+    repeated PropertyValue properties = 4;
+}
+```
+
+The catalog_product smart contract would then be responsible for validating the
+properties field against the CatalogProduct schema at run-time.
+
+A catalog_product entry would reference Grid Product. In addition to
+the properties defined in the CatalogProduct schema. The data model
+would look like this:
+
+```
+Product(
+    product_id="gtin", // Reference from Grid Product
+    product_namespace="GS1" // Reference from Grid Product
+    owner='Target', // Reference from Grid Product
+    properties=[
+        PropertyValue(
+            name="catalog_id",
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="e70fe351d1d096c"
+        ),
+        PropertyValue(
+            name="status",
+            data_type=PropertyDefinition.DataType.ENUM,
+            enum_value=1, # ACTIVE (default)
+        ),
+        PropertyValue(
+            name="price"
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="1-10:$50, 10-50:$40, 50+:$30"  # price windows
+        ),
+        PropertyValue(
+            name="return policy"
+            data_type=PropertyDefinition.DataType.STRING,
+            string_value="30 days from sale"
+        ),
+        # ... Additional Product Properties
+    ])
+```
+
+A catalog product, from a data model perspective, is a grid product. With the 
+differentiators being:
+- It contains the additional fields defined by the catalog_product schema
+- It's stored in the catalog_product state location within the merkle tree
+
+## Transaction Payload and Execution
+
+The following payload contains an action enum and the associated action
+payload.This allows for the action payload to be dispatched to the appropriate 
+logic. Only the defined actions (enum) are available, and only one Action 
+should be performed in a payload.
+
+### CatalogPayload Transaction
+
+```
+message CatalogPayload {
+    enum Actions {
+        UNSET_ACTION = 0;
+        CATALOG_CREATE = 1;
+        CATALOG_UPDATE = 2;
+        CATALOG_DELETE = 3;
+        CATALOG_PRODUCT_CREATE = 100;
+        CATALOG_PRODUCT_UPDATE = 101;
+        CATALOG_PRODUCT_DELETE = 102;
+        CATALOG_PRODUCT_SET_STATUS = 103;
+    }
+
+    Action action = 1;
+
+    // Approximately when transaction was submitted, as a Unix UTC timestamp
+    uint64 timestamp = 2;
+
+    CatalogCreateAction catalog_create = 3;
+    CatalogUpdateAction catalog_update = 4;
+    CatalogDeleteAction catalog_delete = 5;
+    CatalogProductCreateAction catalog_product_create = 6;
```

Comment:
This should jump in number along with the enum, start at 100 or something.

Reply:
Are you suggesting something like this?

```
message CatalogPayload {
    enum Actions {
        UNSET_ACTION = 0;
        CATALOG_CREATE = 1;
        CATALOG_UPDATE = 2;
        CATALOG_DELETE = 3;
        CATALOG_PRODUCT_CREATE = 100;
        CATALOG_PRODUCT_UPDATE = 101;
        CATALOG_PRODUCT_DELETE = 102;
        CATALOG_PRODUCT_SET_STATUS = 103;
    }

    Action action = 1;

    // Approximately when transaction was submitted, as a Unix UTC timestamp
    uint64 timestamp = 2;

    CatalogCreateAction catalog_create = 101;
    CatalogUpdateAction catalog_update = 102;
    CatalogDeleteAction catalog_delete = 103;
    CatalogProductCreateAction catalog_product_create = 104;
    CatalogProductUpdateAction catalog_product_update = 105;
    CatalogProductDeleteAction catalog_product_delete = 106;
    CatalogProductSetStatusAction set_catalog_product_status = 107;
}
```

Reply:
Yes this is what i meant